File: go/pkg/models/friend_invite.go
package models

import (
	"time"

	"github.com/google/uuid"
)

type FriendInvite struct {
	ID        uuid.UUID `gorm:"primaryKey"`
	CreatedAt time.Time
	UpdatedAt time.Time
	InviterID uuid.UUID
	InviteeID uuid.UUID
	Invitee   User
	Inviter   User
}

File: js/packages/types/src/friendInvite.ts
import type { User } from "./user";

type FriendInvite = {
    id: string;
    createdAt: string;
    updatedAt: string;
    inviterId: string;
    inviteeId: string;
    invitee: User;
    inviter: User;
  };
  
  export type { FriendInvite };